### PR TITLE
B4 and H1 Admin Course Request, Integrate Redux into Cart, Fix 4 Tab Responsive view

### PR DESCRIPTION
**Pull request by rawrrencee**
# Feature

Use Case: B.4.1 to B.4.4 (Course Requests)
Feature ID: 8
Feature: Course Content

Use Case: H.1 Cart Integration with Redux
Feature ID: 9
Feature: Purchasing

# Changelog
Removed mentorshipListings() and admin() from Redux
Added cart() to Redux global state
Course Requests table created with filters built into CountIconWidgets (tried it and it was pretty cool to me)
Added overflow-x-scroll to make 4 tabs scroll horizontally (quick fix to retain responsiveness)
Course Requests for admin integrates Student view for viewing courses and lessons
(TODO: test course viewing with backend announcements fix, block commenting for admins)

https://user-images.githubusercontent.com/43084055/111821151-4b945000-891d-11eb-97f0-3e8850be161b.mp4

https://user-images.githubusercontent.com/43084055/111821048-30294500-891d-11eb-9b15-188b4af33682.mov

https://user-images.githubusercontent.com/43084055/111821053-31f30880-891d-11eb-88f5-045c785f091d.mov

## Checklist:

- [x] Merged latest develop
- [x] PR title makes sense

## Screenshots (where relevant):
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/components/Mentorship/ApplicationsTable/index.js [hunk 1]
```diff
@@ -83,16 +82,11 @@ const MentorshipApplicationsTable = () => {
...
-      showNotification('success', SUCCESS, MENTORSHIP_ADD)
-    } else {
-      showNotification('error', UNABLE_ADD, MENTORSHIP_ALR_IN)
-    }
+  const addItemToCart = record => {
+    dispatch({
+      type: 'cart/ADD_MENTORSHIP_LISTING_TO_CART',
+      payload: { mentorshipListingId: record.mentorshipContractId },
+    })
   }
```
Comment (y1bs) [inline src/components/Mentorship/ApplicationsTable/index.js:90]: Maybe wanna add back the showNotification after this? Or do you is it not necessary?

File: src/pages/courses/view/index.js [hunk 2]
```diff
@@ -31,13 +32,13 @@ const ViewCourseDetailsPublic = () => {
...
-    if (response && response.success) {
-      showNotification('success', SUCCESS, response.message)
-    }
+  const addToCart = () => {
+    if (!isEmpty(user.accessToken))
+      dispatch({
+        type: 'cart/ADD_COURSE_TO_CART',
+        payload: { courseId: id },
+      })
+    else history.push('/auth/login')
```
Comment (y1bs) [inline src/pages/courses/view/index.js:41]: Same as above, wanna show notifications too?

**Review by y1bs — COMMENTED** [review]
Also, need to Empty Cart after successful transaction, cuz I realized that the cart is still in the same state, with the old items, after successful checkout.

File: src/pages/admin/courses/index.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  const filterRequests = (requests, adminVerified) => {
+    const preparedData = sortAndKeyRequests(requests, 'desc')
+    if (!isNil(adminVerified)) {
+      return preparedData.filter(o => {
+        return o.adminVerified === adminVerified
+      })
+    }
+    return requests
```
Comment (natalieloke) [inline src/pages/admin/courses/index.js:91]: personally i would separate out `sortAndKey` from `filter` by doing sortAndKey first and then passing the sorted data into the filter method for easier debug in future! 

can consider using lodash's filter which also provides a shorthand next time https://lodash.com/docs/4.17.15#filter
i believe `filter(preparedData, ['adminVerified', adminVerified]` would work

**Review by natalieloke — COMMENTED** [review]
can see if you wanna implement these code-formatting changes; i agree with yibs on the cart state that he mentioned

File: src/components/Mentorship/ApplicationsTable/index.js (hunk 1 above)
Comment (rawrrencee) [reply src/components/Mentorship/ApplicationsTable/index.js:90]: added showMessage and it is done in sagas function!

File: src/pages/courses/view/index.js (hunk 2 above)
Comment (rawrrencee) [reply src/pages/courses/view/index.js:41]: check cart/sagas js file for the notification showing. when u dispatch a method via sagas it is to call an action in the sagas method! I replaced notification with showMessage for this coz like u mentioned, notification  blocks the cart

File: src/pages/admin/courses/index.js (hunk 3 above)
Comment (rawrrencee) [reply src/pages/admin/courses/index.js:91]: ok I will check this. I think coz I wanted a fn that filter + sort + key any array that goes in. so only call once!

**Comment by rawrrencee** [comment]
> can see if you wanna implement these code-formatting changes; i agree with yibs on the cart state that he mentioned

yup addressed what yibs mentioned actly I added into utils.js a showMessage fn that does AntD message instead of notif coz notif blocks the cart

File: src/components/Mentorship/ApplicationsTable/index.js (hunk 1 above)
Comment (y1bs) [reply src/components/Mentorship/ApplicationsTable/index.js:90]: Ahhh okok, ps, missed that

**Review by y1bs — APPROVED** [review]

**Pull request merged**